src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubConfiguration.cs

```diff
@@ -436,6 +475,60 @@ private static byte[] ConvertEventData2Bytes(EventData input)
         private static EventData ConvertString2EventData(string input) 
             => ConvertBytes2EventData(Encoding.UTF8.GetBytes(input));
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA2204:Literals should be spelled correctly", MessageId = "MessagingFactory")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA2204:Literals should be spelled correctly", MessageId = "EventHubClient")]
+        internal static Uri GetEndpointFromEventHubClient(EventHubClient client)
+        {
+            var messagingFactoryProperty = typeof(EventHubClient).GetProperty("MessagingFactory", BindingFlags.Instance | BindingFlags.NonPublic);
+            if (messagingFactoryProperty == null)
+            {
+                throw new InvalidOperationException("Reflection call to obtain EventHubClient.MessagingFactory property info failed. Did a service bus package update occur?");
+            }
+
+            var messagingFactory = (MessagingFactory)messagingFactoryProperty.GetValue(client);
+            if (messagingFactory == null)
+            {
+                throw new InvalidOperationException("Reflection call to obtain value of EventHubClient.MessagingFactory failed. Did a service bus package update occur?");
+            }
+
+            return messagingFactory.Address;
+        }
+
+        internal static string GetLookupKey(Uri endpoint, string hubName)
+            => $"{endpoint.Host}:{hubName}";
+
+        internal static string GetLookupKey(EventHubClient client)
+            => GetLookupKey(GetEndpointFromEventHubClient(client), client.Path);
+
+        internal static string GetLookupKey(string eventHubName, string connectionString)
```

Comment:
>GetLookupKey(string eventHubName [](start = 31, length = 32)

nit: it's very easy to have Uris represented as strings, so it's dangerous to have overloads on Uri vs. String that have different semantics.  Consider giving the overloads different names.